bundles/tampere/content-editor/view/SideContentEditor.js

```diff
@@ -455,28 +546,24 @@ Oskari.clazz.define('Oskari.tampere.bundle.content-editor.view.SideContentEditor
          */
         _fillLayerGeometries: function (geometries) {
             var me = this;
-            if (me.layerGeometries != null && me.layerGeometries.geometry != null) {
-                var layerGeometries = JSON.parse(new olFormatGeoJSON().writeGeometry(me.layerGeometries.geometry));
-                if (me._geojson && me._geojson.features[0] && me._geojson.features[0].geometry) {
-                    layerGeometries = me._geojson.features[0].geometry;
-                }
 
-                if (layerGeometries != null) {
-                    if (layerGeometries.type == 'MultiPoint') {
+            var fillGeometries = function (geom) {
+                if (geom != null) {
+                    if (geom.type === 'MultiPoint') {
                         geometries.type = 'multipoint';
-                        layerGeometries.coordinates.forEach(function (coordinates) {
+                        geom.coordinates.forEach(function (coordinates) {
```

Comment:
Why are the coordinates copied? Could be an own method for that takes the input and returns a clear output instead of modifying an existing variable.

Reply:
Moved to own method.